setup.py

```diff
@@ -69,20 +72,34 @@ class CleanCommand(Clean):
 
     def run(self):
         Clean.run(self)
+        # Remove c files if we are not within a sdist package
+        cwd = os.path.abspath(os.path.dirname(__file__))
+        remove_c_files = not os.path.exists(os.path.join(cwd, 'PKG-INFO'))
+        if remove_c_files:
+            print('Will remove generated .c files')
         if os.path.exists('build'):
             shutil.rmtree('build')
         for dirpath, dirnames, filenames in os.walk('sklearn'):
             for filename in filenames:
-                if (filename.endswith('.so') or filename.endswith('.pyd')
-                        or filename.endswith('.dll')
-                        or filename.endswith('.pyc')):
+                if any(filename.endswith(suffix) for suffix in
+                        (".so", ".pyd", ".dll", ".pyc")):
                     os.unlink(os.path.join(dirpath, filename))
+                if remove_c_files:
+                    extension = os.path.splitext(filename)[1]
+                    if extension in ['.c', '.cpp']:
+                        pyx_file = replace(filename, extension, '.pyx')
+                        if os.path.exists(os.path.join(dirpath, pyx_file)):
+                            os.unlink(os.path.join(dirpath, filename))
             for dirname in dirnames:
                 if dirname == '__pycache__':
                     shutil.rmtree(os.path.join(dirpath, dirname))
+            if remove_c_files:
```

Comment:
Why does it need to be done inside `for dirpath, dirnames, filenames in os.walk('sklearn')`?